vnext/Desktop/Modules/WebSocketModule.cpp

```diff
@@ -148,27 +148,39 @@ std::shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_
   {
     auto ws = IWebSocketResource::Make(std::move(url));
     auto weakInstance = this->getInstance();
-    ws->SetOnError([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()](const IWebSocketResource::Error& err)
+    ws->SetOnError([this, id, weakInstance](const IWebSocketResource::Error& err)
     {
-      auto strongWs = weakWs.lock();
+      auto strongInstance = weakInstance.lock();
+      if (!strongInstance)
+        return;
+
       auto errorObj = dynamic::object("id", id)("message", err.Message);
       this->SendEvent("websocketFailed", std::move(errorObj));
     });
-    ws->SetOnConnect([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()]()
+    ws->SetOnConnect([this, id, weakInstance]()
     {
-      auto strongWs = weakWs.lock();
+      auto strongInstance = weakInstance.lock();
+      if (!strongInstance)
+        return;
+
       auto args = dynamic::object("id", id);
       this->SendEvent("websocketOpen", std::move(args));
     });
-    ws->SetOnMessage([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()](size_t length, const string& message)
+    ws->SetOnMessage([this, id, weakInstance](size_t length, const string& message)
     {
-      auto strongWs = weakWs.lock();
+      auto strongInstance = weakInstance.lock();
+      if (!strongInstance)
+        return;
+
       auto args = dynamic::object("id", id)("data", message)("type", "text");
       this->SendEvent("websocketMessage", std::move(args));
     });
-    ws->SetOnClose([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()](IWebSocketResource::CloseCode code, const string& reason)
+    ws->SetOnClose([this, id, weakInstance](IWebSocketResource::CloseCode code, const string& reason)
```

Comment:
>this [](start = 20, length = 4)

Remove "this"

Reply:
How else can we invoke `SendEvent`, which is a member of `WebSocketModule`?